lib/internal/process/promises.js

```diff
@@ -99,3 +99,12 @@ function setupPromises(scheduleMicrotasks) {
 
   return emitPendingUnhandledRejections;
 }
+
+const savedString = String;
+function safeString(value) {
+  try {
+    return savedString(value);
+  } catch (err) {
+    return '<toString() threw exception>';
```

Comment:
@Fishrock123 A promise can be rejected with any JS value, and inspecting that is not foolproof.

Also, since we have `[Symbol.toStringTag]`, `Object.prototype.toString()` may throw as well.